### ISSUE TITLE
g++ compatibility

### ISSUE THREAD

**Issue by yhaenggi** [Anoncoin/anoncoin#11]
due to 57ebf75e46d6666c52e079afb75e761899a49215 only g++-6 or higher can compile code. this isnt a good idea.

**Comment by yhaenggi** [comment]
please provide ifdefs for g++ versions, @eddef .

**Comment by mikalv** [comment]
Yes will do. Unfortunately a lot of people still runs prehistoric compilers.

**Comment by GroundRod** [comment]
Wow another complex topic,  I'm going to leave this open for a few more days, if anyone cares to comment on current build recommends.

Otherwise its very stale and I'll close it soon.
